### PR TITLE
LVNODCDXI-38

### PR DESCRIPTION
**Pull request by NadiiaChorna**
Hi, guys. 
I've moved all logic of authentication to the one container, and fixed problem with fields clearing on changing form.
Please review my changes. 
Thanks.

### CODE REVIEW

File: src/client/containers/authentication/authentication.js [hunk 1]
```diff
@@ -1,23 +1,55 @@
...
 import './authentication.css';
 
 class Authentication extends Component {
     state = {
       isMember: true,
+      userData: {},
+    };
+
+    onInputChange = ({ target: { name, value } }) => {
+      this.setState(({ userData }) => {
```
Comment (tonypntsrn) [inline src/client/containers/authentication/authentication.js:18]: Are you actually need callback in setState?

File: src/client/components/formComponent/formComponent.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import RenderFormFields from '../renderFormFields/renderFormFields';
+import './formComponent.css';
+
+const FormComponent = ({
+  fieldsToRender,
+  onInputChange,
+  onFormSubmit,
+  userData,
+}) => (
+  <div>
```
Comment (GeFeSTTS) [inline src/client/components/formComponent/formComponent.js]: Please, add className to div container.

Comment (NadiiaChorna) [reply src/client/components/formComponent/formComponent.js]: Sorry, I don't understand why I need the "className" here, could you explain?

Comment (Kindest13) [reply src/client/components/formComponent/formComponent.js]: why do you need to use div in general here? use or fragment or delete it

Comment (NadiiaChorna) [reply src/client/components/formComponent/formComponent.js]: Removed redundant div

**Review by GeFeSTTS — DISMISSED** [review]

File: src/client/containers/authentication/authentication.js (hunk 1 above)
Comment (NadiiaChorna) [reply src/client/containers/authentication/authentication.js:18]: I've been waiting for this question) Well , actually not really but this syntax is more beautiful for me. It takes less code then set state through the object. Do you think it's better to rewrite this?

**Review by tonypntsrn — APPROVED** [review]

**Review by GeFeSTTS — APPROVED** [review]

**Pull request merged**